datafusion/functions/src/core/arrow_cast.rs

```diff
@@ -19,16 +19,50 @@
 //! casting to arbitrary arrow types (rather than SQL types)
 
 use std::{fmt::Display, iter::Peekable, str::Chars, sync::Arc};
+use std::any::Any;
+use arrow::compute::cast;
 
 use arrow_schema::{DataType, Field, IntervalUnit, TimeUnit};
-use datafusion_common::{
-    plan_datafusion_err, DFSchema, DataFusionError, Result, ScalarValue,
-};
+use datafusion_common::{plan_datafusion_err, DataFusionError, Result, ScalarValue, internal_err};
 
-use datafusion_common::plan_err;
-use datafusion_expr::{Expr, ExprSchemable};
+use datafusion_expr::{ColumnarValue, ScalarUDFImpl, Signature, Volatility};
 
-pub const ARROW_CAST_NAME: &str = "arrow_cast";
+#[derive(Debug)]
```

Comment:
this looks just about perfect. Thank you @brayanjuls 

I plan to get https://github.com/apache/arrow-datafusion/pull/8985 merged shortly and then this PR should be possible. 

🙏